### ISSUE TITLE
Can one increase the reported available space on mount?

### ISSUE THREAD

**Issue by datumbox** [s3fs-fuse/s3fs-fuse#2489]
#### Version of s3fs being used (`s3fs --version`)
V1.93 (commit:unknown)

### Details about issue
S3fs is a very handy solution for taking S3 backups using Ubuntu's Deja-dup. As of recently, [they've introduced](https://discourse.gnome.org/t/deja-dup-no-longer-works-with-s3fs/22008) a check to ensure there is sufficient free available space on disk before taking a backup. Given that S3fs mounts the bucket and reports a 4GB free-space, that leads to errors. Example:
```
df -h --total /my/mount/point
Filesystem Size Used Avail Use% Mounted on
s3fs 4.0G 0 4.0G 0% /my/mount/point
total 4.0G 0 4.0G 0% -
```

I wonder if there is currently a way to increase this reported size to avoid issues with some downstream applications such as Deja-dup.

Thanks!

**Comment by gaul** [comment]
Which operating system do you run on?  My Linux system reports 64 PB by default.  The macOS logic differs though:

```
    stbuf->f_blocks = bucket_block_count;
    stbuf->f_frsize = stbuf->f_bsize;
    stbuf->f_bfree  = stbuf->f_blocks;
    stbuf->f_files  = UINT32_MAX;
    stbuf->f_ffree  = UINT32_MAX;
    stbuf->f_favail = UINT32_MAX;
```

You may be able to work around this by passing a larger value to `-o bucket_size` but we should really figure out what the correct default should be.  Maybe macOS reports a strange size for one of the `statvfs` values, e.g., `stbuf->f_bsize` = 1?

**Comment by datumbox** [comment]
Hi @gaul. Thanks so much for your ultra prompt reply.

I'm on Ubuntu 24.04 and I've faced the issue after updating from 22.04.

I've updated my mounting command as you proposed:
`s3fs bucket_name_goes_here /my/mount/point -o bucket_size=64PB`

As you indicated, this indeed updates the reported free size. I've tested the integration with Deja-dup and everything works as expected.

Thank you very much for your help on this. Apologies for raising it as an issue on the main repo, but I hope others will find it useful.
